algorithms/Java/searching/allocate-min-pages.java

```diff
@@ -0,0 +1,77 @@
+/* QUESTION:
+You are given N number of books. Every ith book has pages[i] number of pages. 
+You have to allocate books to M number of students. There can be many ways or permutations to do so. 
+In each permutation, one of the M students will be allocated the maximum number of pages.
+Out of all these permutations, the task is to find that particular permutation in which the maximum number of pages allocated to a student is minimum of those in all the other permutations and print this minimum value.
+Each book will be allocated to exactly one student. Each student has to be allocated at least one book. */
+
+import java.io.*;
+import java.util.*;
+class GFG {
+	public static void main (String[] args) {
+		Scanner sc=new Scanner(System.in);
+		int t=sc.nextInt();  //number of test cases
+		while(t-->0){
+		    int N=sc.nextInt();
+		    int pages[]=new int[n];
```

Comment:
Hi @aayushjain7 , Please check what is `n` in the following: ```java new int[n] ```

Reply:
Oh. I'll correct it.